ggml/src/ggml-vulkan/ggml-vulkan.cpp

```diff
@@ -11443,13 +11443,36 @@ void ggml_backend_vk_get_device_memory(int device, size_t * free, size_t * total
     GGML_ASSERT(device < (int) vk_instance.device_indices.size());
 
     vk::PhysicalDevice vkdev = vk_instance.instance.enumeratePhysicalDevices()[vk_instance.device_indices[device]];
-
     vk::PhysicalDeviceMemoryProperties memprops = vkdev.getMemoryProperties();
 
-    for (const vk::MemoryHeap& heap : memprops.memoryHeaps) {
+    std::vector<vk::ExtensionProperties> extensionprops = vkdev.enumerateDeviceExtensionProperties();
+    vk::PhysicalDeviceMemoryBudgetPropertiesEXT budgetprops;
+    vk::PhysicalDeviceMemoryProperties2 memprops2 = {};
+    bool membudget_extension_supported = false;
+
+    for (const auto & ext : extensionprops) {
+        if (std::string(ext.extensionName.data()) == VK_EXT_MEMORY_BUDGET_EXTENSION_NAME) {
+            membudget_extension_supported = true;
```

Comment:
This list can include hundreds of extensions, I think you should precompute this when the instance is created.